androidApp/src/main/java/com/mbta/tid/mbta_app/android/MainActivity.kt

```diff
@@ -23,14 +25,34 @@ import com.mbta.tid.mbta_app.android.util.LocalLocationClient
 import com.mbta.tid.mbta_app.android.util.fcmToken
 import com.mbta.tid.mbta_app.initializeSentry
 import com.mbta.tid.mbta_app.routes.DeepLinkState
+import kotlin.toString
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 
 class MainActivity : ComponentActivity() {
     private lateinit var fusedLocationClient: FusedLocationProviderClient
+    val deepLinkStateFlow: MutableStateFlow<DeepLinkState?> = MutableStateFlow(null)
+
+    fun handleIntent(intent: Intent?) {
+        val deepLinkUri: Uri? = intent?.data?.takeIf { intent.action == Intent.ACTION_VIEW }
+        deepLinkStateFlow.update {
```

Comment:
Since the new value doesn’t depend on the old value, you can just `deepLinkStateFlow.value =` instead of calling `.update`.